File: js/task3c/S3_PIPE_01.js
function pipe(...fns) {
  return function(initialValue) {
    return fns.reduce((acc, fn) => fn(acc), initialValue);
  };
}
const double = x => x * 2;
const increment = x => x + 1;
const square = x => x * x;
const myPipe = pipe(double, increment, square);
console.log(myPipe(3)); 

File: python/task3c/S3_PIPE_01.py
def pipe(*fns):
  def inner(initial_value):
      result = initial_value
      for fn in fns:
          result = fn(result)
      return result
  return inner

double = lambda x: x * 2
increment = lambda x: x + 1
square = lambda x: x * x

my_pipe = pipe(double, increment, square)
print(my_pipe(3))
